### PR TITLE
chore(auth-node): allow declarative signin resolvers to take precedence

### PR DESCRIPTION
**Pull request by Zaperex**
## Hey, I just made a Pull Request!

This PR modifies the `createOAuthProviderFactory` to allow users to set a default sign in resolver for their auth provider while also allowing them to setup declarative sign in resolvers to override default sign in resolvers.

ex: 
```ts
const authFactory = createOAuthProviderFactory({
  authenticator: customAuthenticator,
  signInResolver: defaultSignInResolver,
  signInResolverFactories: {
    ...declarativeSignInResolvers,
  }
});
```

This allows the `signInResolverFactories` option to take precedence over the `signInResolver` so that users can set defaults if no declarative sign-in resolvers are configured.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by Zaperex** [comment]
Hmm I can't seem to replicate the [failed test in the `auth-backend-module-oidc-providers`](https://github.com/backstage/backstage/actions/runs/9522345816/job/26251757096?pr=25254) package on my local system. Switched to `node 18.19.0`, did a `yarn install --check-cache` before running the unit tests.

**Comment by Zaperex** [comment]
After rerunning the CI, it seems the original package that failed (auth-backend-module-oidc-provider) [passes now](https://github.com/backstage/backstage/actions/runs/9523147789/job/26254063705?pr=25254#step:10:2873), but a unit test from the scaffolder-backend-module-gitlab seemed to be [failing now](https://github.com/backstage/backstage/actions/runs/9523147789/job/26254063705?pr=25254). I'm assuming the tests are just being flaky?

**Comment by Zaperex** [comment]
not stale

**Comment by Zaperex** [comment]
not stale

File: .changeset/neat-socks-cheer.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-auth-node': patch
+---
+
+allow declarative sign in resolvers declared in `signInResolverFactories` to take precedence over the statically defined sign in resolvers in `signInResolver` for the `createOAuthProviderFactory`.
```
Comment (Rugvip) [inline .changeset/neat-socks-cheer.md:5]:
```suggestion
---
'@backstage/plugin-auth-node': minor
---

**BREAKING**: Sign-in resolvers configured via `.signIn.resolvers` now take precedence over sign-in resolvers passed to `signInResolver` option of `createOAuthProviderFactory`. This effectively makes sign-in resolvers passed via the `signInResolver` the default one, which you can then override through configuration.
```

**Pull request merged**